data-prepper-core/src/main/java/org/opensearch/dataprepper/plugin/ExtensionPluginCreator.java

```diff
@@ -0,0 +1,88 @@
+package org.opensearch.dataprepper.plugin;
```

Comment:
Duplicating a class to avoid a circular dependency is a bit problematic. There may be two other options available to us:

1. We could use the [proxy pattern](https://en.wikipedia.org/wiki/Proxy_pattern). In this pattern, one of the dependencies would get a proxy. It would be given the real instance later. I believe Spring supports this, and if so, it should be relatively easy.
2. You could use named beans and have two instances of `PluginCreator`. They would share the same class and code. But, one would have a name like `pluginCreator` and the other would be named `extensionPluginCreator`. This is less ideal because we still end up with two instances of the same object and it takes up more memory.

Reply:
I think the ideal solution is using the proxy. And specifically making a proxy of `PluginConfigurationObservableRegister` when creating the `PluginCreator`.